relay-server/src/http.rs

```diff
@@ -0,0 +1,299 @@
+            Response::Reqwest(response) => {
+                StatusCode::from_u16(response.status().as_u16()).unwrap()
+            }
+        }
+    }
+
+    pub fn json<T: 'static + DeserializeOwned, E>(
+        self,
+        limit: usize,
+    ) -> Box<dyn Future<Item = T, Error = E>>
+    where
+        E: From<reqwest::Error> + From<actix_web::error::JsonPayloadError> + 'static,
+    {
+        // TODO: apply limit to reqwest
+        match self {
+            Response::Actix(response) => {
+                let future = response.json().limit(limit).map_err(From::from);
+                Box::new(future) as Box<dyn Future<Item = _, Error = _>>
+            }
+            Response::Reqwest(response) => {
+                let future = response.json().boxed_local().compat().map_err(From::from);
+                Box::new(future) as Box<dyn Future<Item = _, Error = _>>
+            }
+        }
+    }
+
+    pub fn consume<E>(self) -> ResponseFuture<Self, E>
+    where
+        E: From<actix_web::error::PayloadError> + 'static,
+    {
+        // consume response bodies to allow connection keep-alive
+        match self {
+            Response::Actix(ref response) => Box::new(
+                response
+                    .payload()
+                    .for_each(|_| Ok(()))
+                    .map(|_| self)
+                    .map_err(From::from),
+            ),
+            Response::Reqwest(_) => {
+                // TODO: is this necessary for reqwest?
+                Box::new(future::ok(self))
+            }
+        }
+    }
+
+    pub fn get_header(&self, key: &str) -> Option<&[u8]> {
+        match self {
+            Response::Actix(response) => Some(response.headers().get(key)?.as_bytes()),
+            Response::Reqwest(response) => Some(response.headers().get(key)?.as_bytes()),
+        }
+    }
+
+    pub fn get_all_headers(&self, key: &str) -> Vec<&[u8]> {
+        match self {
+            Response::Actix(response) => response
+                .headers()
+                .get_all(key)
+                .into_iter()
+                .map(|x| x.as_bytes())
+                .collect(),
+            Response::Reqwest(response) => response
+                .headers()
+                .get_all(key)
+                .into_iter()
+                .map(|x| x.as_bytes())
+                .collect(),
+        }
+    }
+
+    pub fn clone_headers(&self) -> Vec<(String, Vec<u8>)> {
+        match self {
+            Response::Actix(response) => response
+                .headers()
+                .iter()
+                .map(|(k, v)| (k.as_str().to_owned(), v.as_bytes().to_owned()))
+                .collect(),
+            Response::Reqwest(response) => response
+                .headers()
+                .iter()
+                .map(|(k, v)| (k.as_str().to_owned(), v.as_bytes().to_owned()))
+                .collect(),
+        }
+    }
+
+    pub fn bytes<E>(self, limit: usize) -> ResponseFuture<Vec<u8>, E>
```

Comment:
Should this be `to_vec`?

Reply:
The name was taken from reqwest API: `response.bytes(..)`

Alternatively from awc: `response.body(..)`

`response.to_vec(..)` sounds alien

Reply:
Would slightly prefer `body()`, but both work for me.